Package evdi as akmod

ffgiff opened the pull request:
Create spec files for evdi driver and displaylink userland module.
Add install test on Travis showing that Centos build fails (#13).
Fixes #81

ssaavedra commented:
Thank you so much for your contribution.

First of all, the Travis test is something that is very interesting for supporting CentOS in the future, so it would be great to have that integrated, even if that is not the core value porposal of this PR.

On the akmod side of things, I'd rather have @kwizart review this here, if possible. I am not an expert on building akmods, and I don't know if there are any idioms or tests that should pass :-)

If @kwizart or the RPMFusion guys require a preliminary version to try this, I could trigger a Travis build for a pre-release branch from this uncommitted PR. Would that be useful?

kwizart commented:
Our review process is very different from all travis/ci , so many files wouldn't be relevant.

Basically best would be to review the spec/src.rpm from bugzilla.rpmfusion.org.
Then, it will be possible to consider to merge back the spec here.

Our review process expect a dedicated bugzilla entry to be created with a spec and src.rpm publication.
Also you need to verify that the packages are building in mock (so there is no dependencies missing others than the one that are considered as a default in the buildroot).

Thx

kwizart commented on evdi-kmod.spec:
```diff
@@ -0,0 +1,71 @@
+
+%changelog
+* Thu May 30 2019 ffgiff <ffgiff@gmail.com>
+        - First version
```
Ident, (use space)

ssaavedra commented on evdi-kmod-common.spec:
```diff
@@ -0,0 +1,94 @@
+/etc/udev/rules.d/99-displaylink.rules
+/etc/X11/xorg.conf.d/20-displaylink.conf
+%dir /usr/libexec/displaylink
+/usr/libexec/displaylink/*
```
I think in this case, if it's possible, it would be nice to have an exception to leave it that way so that we have less problems with forward-compatibility in case displaylink renames or adds new files here, and we miss those on our packaging update (which we might not notice in a CI environment without manual verification).

My opinion here is that hardening against delivering an incomplete release is likely a better trade-off than the alternative, but please feel free to convince me otherwise :-)

kwizart replied:
I understand that from a single package point of view, using wildcard seems convenient. But as a package maintainer, we want to ensure that no unexpected binaries will conflict with another package on end-user "before" they installed on their system.
This can be achieved by listing them explicitly.
So something like displaylink* is appropriate, but display* (it may conflicts with ImageMagick  package that has a display binary among others).

ssaavedra replied:
In this case we are already inside `/usr/libexec/displaylink`, that's why I thought it would be *safe enough*. The case I can think of is that the user manually installed displaylink before installing the RPM, but that is an expected clash. Other packages should not be expected to need the `/usr/libexec/displaylink` folder, or should we not make that assumption?

kwizart replied:
Using libexec/displaylink is a good idea, specially for content that isn't supposed to be used by end-users directly. Should we expect some binaries to be used by end-users ?
Is this directory relevant for all binaries ?

ssaavedra commented:
Thank you so much for this review and for the interest in contributing this to RPMFusion!

ffgiff commented on evdi-kmod-common.spec:
```diff
@@ -0,0 +1,94 @@
+Source2:        99-displaylink.rules
+Source3:        displaylink-sleep-extractor.sh
+# From http://www.displaylink.com/downloads/ubuntu.php
+Source4:	DisplayLink USB Graphics Software for Ubuntu %{_daemon_version}.zip
```
You mean so there should be three spec files: evdi, evdi-kmod and displaylink? That would work, but I'd need to move the DKMS part of displaylink.spec into a submodule to avoid the obvious clash.

kwizart replied:
RPM Fusion do not support dkms method, but you can support it if you like to (it would be a separate spec file).
We don't rely on your SCM at all if the spec are imported into RPM Fusion, it will have it's separate life.
You can have all spec into a "rpm" sub-directory of your project.

ffgiff commented on evdi-kmod-common.spec:
```diff
@@ -0,0 +1,94 @@
+# From http://www.displaylink.com/downloads/ubuntu.php
+Source4:	DisplayLink USB Graphics Software for Ubuntu %{_daemon_version}.zip
+Source5:        20-displaylink.conf
+License:       GPLv2.1, MIT and Proprietary
```
Yes, it's also wrong, it should be *L*GPLv2.1 for libevdi, with the MIT and proprietary bits being the bits in displaylink.spec

ffgiff commented:
Thanks for the review, I'll update the PR in a week or so once I can get around to addressing all the issues. I think it does deserve some changes to be made in displaylink.spec, at least the dkms and libevdi parts need to be submodules so they won't clash with the akmod. That should make the licensing clearer too, but initially I was aiming to keep the change as unobtrusive as possible.

I'll also see about making a separate PR for the travis change.

ffgiff commented on evdi-kmod-common.spec:
```diff
@@ -0,0 +1,94 @@
+/etc/X11/xorg.conf.d/20-displaylink.conf
+%dir /usr/libexec/displaylink
+/usr/libexec/displaylink/*
+%dir /var/log/displaylink/
```
I assume the logs are written by the DisplayLinkManager process which is run by systemd. The logs are written as root.

ffgiff commented:
Reopening this. I've been happily running my own akmods for well over a year now. Notwithstanding the current issues around the evdi 1.9.1 tagging, I'd like to get the ball rolling on this again and ultimately get it into rpmfusion.

ffgiff commented:
Replaced by #166 as the rework has confused GitHub which produces diffs to the wrong commit on the master branch.
